@here/harp-mapview/lib/geometry/overlayOnElevation.ts

```diff
@@ -20,6 +29,58 @@ function overlayObject(object: TileObject, displacementMap: THREE.DataTexture):
     }
 }
 
+const worldCoords = new THREE.Vector3();
+
+/**
+ * Overlays the specified coordinates on top of elevation data if available.
+ *
+ * @param tileCoords World coordinates relative to the `tile` center.
+ * @param elevationProvider Used to get elevation data.
+ * @param tile The tile to which the coordinates are relative.
+ */
+function overlayPosition(
+    tileCoords: THREE.Vector3,
+    elevationProvider: ElevationProvider,
+    tile: Tile
+) {
+    worldCoords.copy(tileCoords).add(tile.center);
+    const geoCoords = tile.mapView.projection.unprojectPoint(worldCoords);
+    const height = elevationProvider.getHeight(geoCoords);
+
+    if (height !== undefined) {
+        geoCoords.altitude = height;
+        tile.mapView.projection.projectPoint(geoCoords, worldCoords);
+        tileCoords.copy(worldCoords).sub(tile.center);
+    }
+}
+
+/**
+ * Overlays a text element on top of elevation data if available.
+ *
+ * @param textElement The text element whose geometry will be overlaid.
+ * @param elevationProvider Used to get elevation data.
+ * @param tile The tile the text element belongs to.
+ */
+function overlayTextElement(
+    textElement: TextElement,
+    elevationProvider: ElevationProvider,
+    tile: Tile
+) {
+    // TODO: Move calculation of text element geoCoordinates to decoder.
+    if (textElement.path === undefined) {
+        overlayPosition(textElement.position, elevationProvider, tile);
+        return;
+    }
+
+    for (const position of textElement.path) {
+        overlayPosition(position, elevationProvider, tile);
+    }
+}
+
+// Overlay of text elements within a tile is spread through this many frames to minimize the time
+// spent per frame on elevation overlay.
+const TEXT_OVERLAY_SPREAD_FRAME_COUNT = 60;
+
 /**
  * Overlays the geometry in the given tile on top of elevation data if available.
  *
```

Comment:
There is a property like "fadingEnabled" in MapView. If fading is disabled, I think that all labels should be placed at once.
Also, it has to be checked that the FRAME_COMPLETE event is sent _after_ all labels have been set on the "overlay".

Reply:
there is a update requested everytime we update the label overlay, that should delay the FRAME_COMPLETE event.